pymeasure/instruments/scientificmagnetics/scimag.py

```diff
@@ -0,0 +1,448 @@
+        preprocess_reply=extract_value_magnet_status
+    )
+
+    current_status = Instrument.measurement(
+        "K",
+        "Returns dict of current status.",
+        preprocess_reply=extract_value_current_status
+    )
+
+    operating_parameters = Instrument.measurement(
+        "O",
+        "Returns dict of operation parameters",
+        preprocess_reply=extract_value_operating_parameters
+    )
+
+    set_point_status = Instrument.measurement(
+        "S",
+        "Returns dict of set point status.",
+        preprocess_reply=extract_value_set_point_status
+    )
+
+    def __init__(self, resourceName,
+                 ramp_rate_max=5,
+                 voltage_limit_max=1,
+                 field_max=10,
+                 **kwargs):
+        super().__init__(
+            resourceName,
+            "Scientific Magnetics SMC",
+            includeSCPI=False,
+            **kwargs
+        )
+        self._RAMP_RATE_MAX = ramp_rate_max  # A/s
+        self._VOLTAGE_LIMIT_MAX = voltage_limit_max  # V
+        self._T_MAX = field_max  # T
+
+    @property
+    def tesla(self) -> float:
+        """
+        High-Level property for setting and getting the field in the solenoid.
+
+        If installed, the setter-funktion toggles the
+        reversal-switch automatically at zero-field.
+        """
+        self.unit = "T"
+        value = self.output_parameters_tesla["field"]
+        return value
+
+    @tesla.setter
+    def tesla(self, value: float):
+        self.unit = "T"
+        if value == 0:
+            self.ramp_target = "Z"
+        elif abs(value) < self._T_MAX:
+            # 1. Get actual direction
+            if self.operating_parameters["switch_direction"] == 0:
+                sign = +1
+            elif self.operating_parameters["switch_direction"] == 1:
+                sign = -1
+            else:
+                sign = 0
+
+            # 2. Check, if value and sign are both positive or both negative
+            if value * sign > 0:
+                self.upper_setpoint = abs(value)
+            elif value * sign < 0:
+                self.ramp_target = "Z"
+                # At zero-field: switch direction.
+                while not (self.current_status["ramp_status"] == 1):
+                    time.sleep(1)
+                if value > 0:
+                    self.reverse_switch_on = False
+                elif value < 0:
+                    self.reverse_switch_on = True
+                self.upper_setpoint = abs(value)
+            self.ramp_target = "U"
+        else:
+            log.warning("Intended field was: %08.4f T.\nAllowed maximum field is: %08.4f T" % (value, self._T_MAX))
+            raise UserWarning("Intended field value exceeded range of the solenoid. Value was NOT set.")
+
+    def shutdown(self):
+        """Brings the instrument to a safe and stable state.
+        """
+        self.isShutdown = True
+        log.info("Shutting down %s" % self.name)
```

Comment:
I think it is better to use `super().shutdown()` rather than copying the lines of code.